modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -93,46 +93,63 @@ MultiSmoothSuperellipsoidIC::initialSetup()
 void
 MultiSmoothSuperellipsoidIC::computeSuperellipsoidSemiaxes()
 {
-  for (unsigned int i = 0; i < _numbub[_gk]; i++)
+  Real randnum;
+  unsigned int start = _as.size();
+  _as.resize(start + _numbub[_gk]);
+  _bs.resize(start + _numbub[_gk]);
+  _cs.resize(start + _numbub[_gk]);
+
+  for (unsigned int i = start ; i < _as.size(); i++)
   {
-    //Vary bubble radius
     switch (_semiaxis_variation_type)
     {
-    case 0: //Uniform distrubtion
-      _as.push_back(_semiaxis_a[_gk]*(1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_a_variation[_gk]));
-      _bs.push_back(_semiaxis_b[_gk]*(1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_b_variation[_gk]));
-      _cs.push_back(_semiaxis_c[_gk]*(1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_c_variation[_gk]));
-      break;
-    case 1: //Normal distribution
-      _as.push_back(_random.randNormal(_tid, _semiaxis_a[_gk],_semiaxis_a_variation[_gk]));
-      _bs.push_back(_random.randNormal(_tid, _semiaxis_b[_gk],_semiaxis_b_variation[_gk]));
-      _cs.push_back(_random.randNormal(_tid, _semiaxis_c[_gk],_semiaxis_c_variation[_gk]));
-      break;
-    case 2: //No variation
-      _as.push_back(_semiaxis_a[_gk]);
-      _bs.push_back(_semiaxis_b[_gk]);
-      _cs.push_back(_semiaxis_c[_gk]);
-    }
-
-    if (_as.back() < 0.0) _as.back() = 0.0;
-    if (_bs.back() < 0.0) _bs.back() = 0.0;
-    if (_cs.back() < 0.0) _cs.back() = 0.0;
+      case 0: //Uniform distrubtion
+        randnum = _random.rand(_tid);
+        _as[i] = _semiaxis_a[_gk] * (1.0 + (1.0 - 2.0 * (_vary_axes_independently ? _random.rand(_tid) : randnum)) * _semiaxis_a_variation[_gk]);
```

Comment:
no ternary operator here! Just use the value of rand num. You are currently getting **4** random numbers for the independent case!